src/main/java/staffconnect/model/availability/Availability.java

```diff
@@ -0,0 +1,87 @@
+package staffconnect.model.Availability;
+
+import static java.util.Objects.requireNonNull;
+import static staffconnect.commons.util.AppUtil.checkArgument;
+
+import java.time.DayOfWeek;
+/**
+ * Represents a Person's availability in the staff book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidAvailability(String)}
+ */
+public class Availability {
+
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Should be a week of the day, the full word or the first syllable of the word";
+    public static final String VALIDATION_REGEX = "(?i)((mon|tue(s)?|wed(nes)?|thu(r)?(rs)?|fri|sat(ur)?|sun)(day)?)";
+
+    public final String value;
+
+    /**
+     * Constructs a {@code Availability}.
+     *
+     * @param availability A valid availability.
+     */
+    public Availability(String availability) {
+        requireNonNull(availability);
+        checkArgument(isValidAvailability(availability), MESSAGE_CONSTRAINTS);
+        value = strToDayOfWeek(availability);
+    }
+
+    /**
+     * Returns true if a given string is a valid availability.
+     */
+    public static boolean isValidAvailability(String test) {
+        return test.matches(VALIDATION_REGEX);
+    }
+
+    /**
+     * Returns a standardised Day of the Week.
+     *
+     * @param validAvailability validatedAvailability to be translated.
+     * @return Corresponding Day of Week.
+     */
+    public static String strToDayOfWeek(String validAvailability) {
```

Comment:
The regular expression is great, but is it every possible to use a switch case statement rather than else-ifs. I am not familiar with so just a suggestion.

Reply:
Based on this, i don't think I can use regex for switch case.
https://stackoverflow.com/questions/8088370/using-regex-for-switch-statement-in-java